### ISSUE TITLE
How to connect the node like A111 "ControlNet is more important"

### ISSUE THREAD

**Issue by whl0070179** [Kosinkadink/ComfyUI-Advanced-ControlNet#87]
![微信图片_20240321095253](https://github.com/Kosinkadink/ComfyUI-Advanced-ControlNet/assets/4544479/4de0a176-6032-442a-8f66-93c827bf43a2)

**Comment by Kosinkadink** [comment]
You can't at the moment, but it's on my list of features to work on as I've gotten a lot of requests for it. I'll let you know when it's ready.

**Comment by whl0070179** [comment]
3Q 👍

**Comment by Kosinkadink** [comment]
The "ControlNet is more important" feature is now available. It is exposed via the ```uncond_multiplier``` param on Soft Weights. Setting it to 0.0 is the exact replication of "ControlNet is more important", but with the way it works, it can be tweaked with any value between 0.0 and 1.0, so it is a further extension of the auto1111 sd-webui-controlnet feature. In diffusers, it is referred to as "guess_mode".

(NOTE, when set to 1.0, the Soft Weights are equivalent to "My prompt is more important", same as before.)

![image](https://github.com/Kosinkadink/ComfyUI-Advanced-ControlNet/assets/7365912/096d10d7-5bc1-4489-aec9-4bda33b3f5cf)

**Comment by curlysasha** [comment]
> The "ControlNet is more important" feature is now available. It is exposed via the `uncond_multiplier` param on Soft Weights. Setting it to 0.0 is the exact replication of "ControlNet is more important", but with the way it works, it can be tweaked with any value between 0.0 and 1.0, so it is a further extension of the auto1111 sd-webui-controlnet feature. In diffusers, it is referred to as "guess_mode".
> 
> (NOTE, when set to 1.0, the Soft Weights are equivalent to "My prompt is more important", same as before.)
> 
> ![image](https://private-user-images.githubusercontent.com/7365912/331412357-096d10d7-5bc1-4489-aec9-4bda33b3f5cf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU5MjYzNDksIm5iZiI6MTcxNTkyNjA0OSwicGF0aCI6Ii83MzY1OTEyLzMzMTQxMjM1Ny0wOTZkMTBkNy01YmMxLTQ0ODktYWVjOS00YmRhMzNiM2Y1Y2YucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MTdUMDYwNzI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzgxNWJjOTdlOWYyOTZlNDM3YjJkNmI1NzU2NmVmMzliNDYzNjMyNTZmNDQwNGUwNTk2YWJkNGRlM2U2YWEwMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.t0Bk-DsOkCc43Hmj9OHIyL3kwsFJLfsy9nQGfpXSgAA)

Thank you!

**Comment by whl0070179** [comment]
Thx!

**Comment by wuliwalaboy** [comment]
@Kosinkadink 
when i set  uncond_multiplier= 0.00 as example , it doesnot work
![image](https://github.com/Kosinkadink/ComfyUI-Advanced-ControlNet/assets/131651863/cf3c5bda-27ab-4004-9838-9915de3983cf)

get errors like below:
 cond_or_uncond = self.batched_number.cond_or_uncond
AttributeError: 'int' object has no attribute 'cond_or_uncond'

pls help me ,thx!

**Comment by Kosinkadink** [comment]
Which KSampler are you attempting to use it with?

**Comment by wuliwalaboy** [comment]
![image](https://github.com/Kosinkadink/ComfyUI-Advanced-ControlNet/assets/131651863/db40ed3d-0acd-47cf-bf51-82148ce987f6)
I used both KSamplers, and both gave this error.

**Comment by Kosinkadink** [comment]
Create the simplest workflow that replicates the issue that uses KSampler advanced, and send it my way

**Comment by Kosinkadink** [comment]
I've confirmed that uncond_multiplier still works as intended on my end on latest ComfyUI and Advanced-ControlNet, so perhaps there is something in your workflow that makes the code not go through the usual ComfyUI codepath for sampling. Once you give me the simplest workflow that you can make that still has the issue, that should clear things up.

**Comment by wuliwalaboy** [comment]
![image](https://github.com/Kosinkadink/ComfyUI-Advanced-ControlNet/assets/131651863/ed1a385d-21ee-4d9b-abf1-84313b020811)

I re-experimented and found that when I connect the tiled diffusion node, this error occurs. Here is a partial screenshot of my workflow, please take a look.

**Comment by Kosinkadink** [comment]
Yep, it's likely that. Tiled Diffusion has its own version of sampling code. Currently, the implementation for uncond_multiplier requires that the sampling code is written in a particular manner, as I need to figure out the cond and uncond portions of the batches. I will talk with comfy about changing the controlnet code to provide the additional info, although that will be a change that will require other sampling code (like tiles Diffusion) to account for it anyway.

TL;Dr nothing I can do at the moment to make it work with tiled Diffusion, unless tiled diffusion code slightly modified their sampling code

**Comment by wuliwalaboy** [comment]
I'm really looking forward to it. Thank you again for your response!

**Comment by dushunpeng163** [comment]
> Yep, it's likely that. Tiled Diffusion has its own version of sampling code. Currently, the implementation for uncond_multiplier requires that the sampling code is written in a particular manner, as I need to figure out the cond and uncond portions of the batches. I will talk with comfy about changing the controlnet code to provide the additional info, although that will be a change that will require other sampling code (like tiles Diffusion) to account for it anyway.
> 
> TL;Dr nothing I can do at the moment to make it work with tiled Diffusion, unless tiled diffusion node slightly modifies it's sampling code

hi，@Kosinkadink
May I ask if there has been any progress? I have encountered the same issue. pls help!😢
